raspberry/bzzz/sensors/time_of_flight_sensor.py

```diff
@@ -0,0 +1,196 @@
+        # Latest measurment update variables.
+        self._current_altitude = None
+        self._altitude_readings_list = None
+        self._altitude_readings_list_current_index = None
+        self._num_latest_readings_to_keep = num_latest_readings_to_keep
+
+        # previous measurement varaibles.
+        self._previous_altitude = None
+        self._last_update_time = None
+        self.update_measurement_at_fixed_rate = update_measurement_at_fixed_rate
+
+        # Data cache varaibles.
+        self.cache_altitude = cache_altitude
+        self._altitude_cache = []
+
+        # outlier detection varaibles.
+        self.use_outlier_detection = use_outlier_detection
+        self.__abs_outlier_diff_thres = abs_outlier_diff_thres
+
+        # Call for sensor initialization.
+        self.__init_ToF_sensor()
+
+    def __init_ToF_sensor(self,
+                          address=0x29,
+                          mode=VL53L0X.Vl53l0xAccuracyMode.LONG_RANGE):
+        """Function to initialize the ToF sensor.
+
+        :param address: ToF sensor physical i2C address, defaults to 0x29
+        :param mode: Sensor mode of operation, defaults to VL53L0X.Vl53l0xAccuracyMode.LONG_RANGE
+        """
+        print("Init ToF, wait.....")
+        self.tof = VL53L0X.VL53L0X(i2c_bus=1, i2c_address=address)
+        self.tof.open()
+        self.tof.start_ranging(mode=mode)
+        self.timing = self.tof.get_timing()
+        if self.timing < 20000:
+            self.timing = 20000
+        self._last_update_time = time_ns()
+        self._altitude_readings_list = list()
+        self._altitude_readings_list_current_index = 0
+        for _ in range(self._num_latest_readings_to_keep):
+            self._altitude_readings_list.append(
+                self.__get_current_ToF_measurement())
+            sleep(self.timing/1000000.0)
+        self.__update_altitude()
+        print("Init ToF, Done!")
+
+    @property
+    def abs_outlier_diff_thres(self):
+        """return the current setting of absolute outlier difference threshold.
+
+        :return: absolute outlier difference threshold.
+        """
+        return self.__abs_outlier_diff_thres
+
+    @abs_outlier_diff_thres.setter
+    def abs_outlier_diff_thres(self, threshold):
+        """Setter for absolute outlier difference threshold
+
+        :param threshold: new threshold value.
+        """
+        self.__abs_outlier_diff_thres = threshold
+
+    @property
+    def altitude(self):
+        """Updates and returns the current altitude measurement in mm.
+
+        :return: If there is no measurement error the function returns the current altitude measurement in mm else -1.
+        """
+        status = self.__update_altitude()
+        if status == -1:
+            return -1
+        return self._current_altitude
+
+    @altitude.setter
+    def altitude(self, val):
+        """Setter for altitude measurement.
+        NOTE: only use this for testing purposes, do not manually 
```

Comment:
should this method be removed?

Reply:
Yes, it has no use now. I will remove it.